hw/xbox/nv2a/nv2a.h

```diff
@@ -21,6 +21,24 @@
 #ifndef HW_NV2A_H
 #define HW_NV2A_H
 
+/*These types are declared here, since nv2a.h is a common library and removes the need of 
+  making C++ compatible the structures NV2AState and SurfaceBinding, by exporting everything from the
+  pgraph_upload_surface function*/
+ typedef uint64_t hwaddr;
```

Comment:
Make sure you fix indentation and excess whitespace everywhere before you try to do a final PR.